### DISCUSSION TITLE
PostgreSQL CUBE Data Type example in Milvus

### DISCUSSION THREAD

**Discussion by batara666** [milvus-io/milvus#6201] [Q&A and General discussion] [answered]
I want to migrate PostgreSQL & CUBE Data Type to Milvus, but is Milvus able to handle that kind of data? Thankyou

**Comment by yhmo** [comment, marked as answer]
Milvus is a vector search database. Each vector is a float array like this: [1.0, 2.0, 3.0, ......, 100.0], user can insert million of such kind of vectors into milvus. Then user can give a vector to search, milvus will return the most similar vectors. 
The PostgreSQL CUBE is a SQL statement, to group the sql result by column, like this:
`SELECT c1, c2, sum(c3) FROM table_name GROUP BY CUBE (c1, c2);`

Seem the CUBE result is not like a vector, could you give more information about your scenario?

**Reply by batara666** [reply]
Sorry, I exploited the CUBE feature to works like vector-search database, the replacement is FLOAT_VECTOR data type, thankyou

**Reply by yhmo** [reply]
Anyway, milvus require the insert vectors to be float array, so to migrate the data, you need to:
1. extract FLOAT_VECTOR from postgresql, store to binary file or text file
2. write a program to read the binary file or text file, you can choose python/java/go/c++
3. call milvus sdk api to insert vectors into milvus

**Reply by batara666** [reply]
Sadly, I got problem with Golang SDK

**Reply by yhmo** [reply]
> Sadly, I got problem with Golang SDK

The problem is?

**Reply by batara666** [reply]
@yhmo thankyou for your time, please check the issue section 🙏🙏

**Reply by yhmo** [reply]
> @yhmo thankyou for your time, please check the issue section

You are using milvus v2.0, the v2.0 currently only provide python sdk.
